plugins/vpp/l3plugin/arp_config_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l3plugin_test
+
+import (
+	"git.fd.io/govpp.git/adapter/mock"
+	"git.fd.io/govpp.git/core"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/vpp-agent/idxvpp/nametoidx"
+	"github.com/ligato/vpp-agent/plugins/vpp/binapi/ip"
+	"github.com/ligato/vpp-agent/plugins/vpp/ifplugin/ifaceidx"
+	"github.com/ligato/vpp-agent/plugins/vpp/l3plugin"
+	"github.com/ligato/vpp-agent/plugins/vpp/model/l3"
+	"github.com/ligato/vpp-agent/tests/vppcallmock"
+	. "github.com/onsi/gomega"
+	"testing"
+)
+
+// Test ArpConfigurator initialization
+func TestArpConfiguratorInit(t *testing.T) {
+	RegisterTestingT(t)
+	ctx := &vppcallmock.TestCtx{
+		MockVpp: &mock.VppAdapter{},
+	}
+	connection, _ := core.Connect(ctx.MockVpp)
+	defer connection.Disconnect()
+
+	plugin := &l3plugin.ArpConfigurator{}
+
+	// Test init
+	err := plugin.Init(logging.ForPlugin("test-log", logrus.NewLogRegistry()), connection, nil, false)
+	Expect(err).To(BeNil())
+	err = plugin.Close()
+	Expect(err).To(BeNil())
+}
+
+var arpTableEntries = []l3.ArpTable_ArpEntry{
+	{
+		Interface:   "tap1",
+		IpAddress:   "192.168.10.21",
+		PhysAddress: "59:6C:DE:AD:00:01",
+		Static:      true,
+	},
+	{
+		Interface:   "tap2",
+		IpAddress:   "192.168.10.22",
+		PhysAddress: "59:6C:DE:AD:00:02",
+		Static:      true,
+	},
+	{
+		Interface:   "tap3",
+		IpAddress:   "dead::01",
+		PhysAddress: "59:6C:DE:AD:00:03",
+		Static:      false,
+	},
+	{
+		Interface:   "tap4",
+		IpAddress:   "dead::02",
+		PhysAddress: "59:6C:DE:AD:00:04",
+		Static:      false,
+	},
+}
+
+// Test adding of ARP entry
+func TestAddArp(t *testing.T) {
+	// Setup
+	ctx, connection, plugin := arpTestSetup(t)
+	defer arpTestTeardown(connection, plugin)
+
+	err := plugin.AddArp(&arpTableEntries[2])
+	Expect(err).To(BeNil())
+
+	ctx.MockVpp.MockReply(&ip.IPNeighborAddDelReply{})
+	err = plugin.AddArp(&arpTableEntries[0])
+	Expect(err).To(BeNil())
+
+	// Test isValidARP
+	err = plugin.AddArp(&l3.ArpTable_ArpEntry{})
+	Expect(err).To(Not(BeNil()))
```

Comment:
`Expect(err).To(HaveOccurred())` will be more clear. Having `BeNil()` in both error/non-error cases makes it little harder to read.